### PR TITLE
Support some MPD attributes to encode/decode

### PR DESCRIPTION
**Pull request by miyukki**
- Support new MPD attributes. (`minimumUpdatePeriod `, `availabilityStartTime `)
  - Instead of PRs: https://github.com/zencoder/go-dash/pull/50 https://github.com/zencoder/go-dash/pull/47
- `NewMPD` be able to receive arguments to set the attributes.
- Fix some tests.

### CODE REVIEW

File: mpd/mpd.go [hunk 1]
```diff
@@ -218,19 +263,31 @@ type AudioChannelConfiguration struct {
 
 // Creates a new MPD object.
 // profile - DASH Profile (Live or OnDemand).
-// mediaPresentationDuration - Media Presentation Duration (i.e. PT6M16S).
 // minBufferTime - Min Buffer Time (i.e. PT1.97S).
-func NewMPD(profile DashProfile, mediaPresentationDuration string, minBufferTime string) *MPD {
+// attributes - Other attributes (optional)
+func NewMPD(profile DashProfile, minBufferTime string, attributes ...AttrMPD) *MPD {
```
Comment (bcasinclair) [inline mpd/mpd.go]: Note this makes this a breaking change for any downstream usages

Comment (miyukki) [reply mpd/mpd.go]: Yes, I know. Should I use another method name?

Comment (bcasinclair) [reply mpd/mpd.go]: Also from ISO23009 mediaPresentationDuration is a mandatory field for type "static" which is the default here.

File: mpd/mpd.go [hunk 2]
```diff
@@ -218,19 +263,31 @@ type AudioChannelConfiguration struct {
...
+
+	for i := range attributes {
+		switch attr := attributes[i].(type) {
+		case *attrAvailabilityStartTime:
+			mpd.AvailabilityStartTime = attr.GetStrptr()
+		case *attrMinimumUpdatePeriod:
+			mpd.MinimumUpdatePeriod = attr.GetStrptr()
+		case *attrMediaPresentationDuration:
+			mpd.MediaPresentationDuration = attr.GetStrptr()
+		}
```
Comment (bcasinclair) [inline mpd/mpd.go]: Nice way to add attributes, should add some validation too and along with that some negative test cases

File: mpd/mpd.go (hunk 1 above)
Comment (miyukki) [reply mpd/mpd.go]: Exactly, you are correct. I found this method creates only static type.

Is it a solution that uses two methods?

Example
```go
func NewMPD(profile DashProfile, mediaPresentationDuration, minBufferTime string, attributes ...AttrMPD) *MPD
```

```go
func NewMPDDynamic(profile DashProfile, minBufferTime string, attributes ...AttrMPD) *MPD
```

Comment (miyukki) [reply mpd/mpd.go]: or `NewDynamicMPD` ?

**Comment by miyukki** [comment]
I pushed a new commit, fixed below points:

- Add `NewDynamicMPD` function to generate dynamic type MPD object.
- `NewMPD` and `NewDynamicMPD` need mandatory fields. it be able to receive additional attributes too.
```go
func NewMPD(profile DashProfile, mediaPresentationDuration, minBufferTime string, attributes ...AttrMPD) *MPD 
```
```go
func NewDynamicMPD(profile DashProfile, availabilityStartTime, minBufferTime string, attributes ...AttrMPD) *MPD
```
- Each methods process the additional attributes only supported by specification.

File: mpd/mpd.go (hunk 2 above)
Comment (miyukki) [reply mpd/mpd.go]: Is the validation you are talking about attributes that do not match type, or about the format of attribute values?

**Comment by miyukki** [comment]
@bcasinclair Do you have any questions?

**Pull request merged**